packages/server/src/core/middleware.ts

```diff
@@ -143,27 +156,24 @@ export type MiddlewareFunction<
 };
 
 export function createMiddlewareFactory<TConfig extends AnyRootConfig>() {
-  function createNewBuilder(
-    previousMiddleware: MiddlewareFunction<any, any>,
-    newMiddleware: MiddlewareFunction<any, any>,
-  ) {
-    newMiddleware.piped = [
```

Comment:
Missed this on the first go, but this is a big no-no.

We should never mutate objects that we don't have full control of within the library. This is mutating a user-passed-in function, so we'd be getting very hard-to-diagnose issues if/when someone reused a fn across several pipes